src/Xrefcheck/Verify.hs

```diff
@@ -499,13 +525,8 @@ checkExternalResource VerifyConfig{..} link
               _ -> Left . ExternalHttpResourceUnavailable $ responseStatus resp
           other -> Left . ExternalResourceSomeError $ show other
       where
-        retryAfterInfo :: Response a -> Time Second
-        retryAfterInfo response =
-          fromMaybe vcDefaultRetryAfter $ do
-            retryAfterByteString <- L.lookup hRetryAfter $ responseHeaders response
-            retryAfterString <- rightToMaybe $ decodeUtf8Strict @String retryAfterByteString
-            retryAfter <- readMaybe @Natural retryAfterString
-            pure . sec $ fromIntegral @Natural @RatioNat retryAfter
+        retryAfterInfo :: Response a -> Maybe RetryAfter
+        retryAfterInfo = decodeThrow <=< L.lookup hRetryAfter . responseHeaders
```

Comment:
I don't think we want to use a YAML decoder for this, the header is in plain text.